database/worker.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package database
+
+import (
+	"database/sql"
+
+	"github.com/go-vela/types/library"
+)
+
+// Worker is the database representation of a worker.
+type Worker struct {
+	ID            sql.NullInt64  `sql:"id"`
+	Hostname      sql.NullString `sql:"hostname"`
+	Path          sql.NullString `sql:"path"`
+	Online        sql.NullBool   `sql:"online"`
+	LastCheckedIn sql.NullTime   `sql:"last_checked_in"`
+}
+
+// Nullify ensures the valid flag for
+// the sql.Null types are properly set.
+//
+// When a field within the Build type is the zero
+// value for the field, the valid flag is set to
+// false causing it to be NULL in the database.
+func (w *Worker) Nullify() *Worker {
+	if w == nil {
+		return nil
+	}
+
+	// check if the ID field should be false
+	if w.ID.Int64 == 0 {
+		w.ID.Valid = false
+	}
+
+	// check if the Hostname field should be false
+	if len(w.Hostname.String) == 0 {
+		w.Hostname.Valid = false
+	}
+
+	// check if the URL field should be false
+	if len(w.Path.String) == 0 {
+		w.Hostname.Valid = false
+	}
+
+	return w
+}
+
+// ToLibrary converts the Worker type
+// to a library Worker type.
+func (w *Worker) ToLibrary() *library.Worker {
+	worker := new(library.Worker)
+
+	worker.SetID(w.ID.Int64)
+	worker.SetHostname(w.Hostname.String)
+	worker.SetPath(w.Path.String)
+	worker.SetOnline(w.Online.Bool)
+	worker.SetLastCheckedIn(w.LastCheckedIn.Time)
+	return worker
+}
+
+// WorkerFromLibrary converts the library worker type
+// to a database worker type.
+func WorkerFromLibrary(w *library.Worker) *Worker {
+	worker := &Worker{
+		ID:            sql.NullInt64{Int64: w.GetID(), Valid: true},
+		Hostname:      sql.NullString{String: w.GetHostname(), Valid: true},
+		Path:          sql.NullString{String: w.GetPath(), Valid: true},
+		Online:        sql.NullBool{Bool: w.GetOnline(), Valid: true},
+		LastCheckedIn: sql.NullTime{Time: w.GetLastCheckedIn(), Valid: true},
+	}
+
+	return worker.Nullify()
+}
```

Comment:
I think you'll want to add a `Validate()` function similar to what we have for the other types 👍 

The purpose of these functions is to quickly validate the required fields for the struct have content before creating/updating in the database.

Some examples:

https://github.com/go-vela/types/blob/3e7f85be5d18e6fe39e992946a2b7eaa903800d7/database/build.go#L263-L277

https://github.com/go-vela/types/blob/3e7f85be5d18e6fe39e992946a2b7eaa903800d7/database/hook.go#L139-L158

https://github.com/go-vela/types/blob/3e7f85be5d18e6fe39e992946a2b7eaa903800d7/database/repo.go#L160-L194

Reply:
I went back and forth on this so thanks for mentioning it.

I had looked at those examples and noticed that only required fields are validated. In this case, what fields would be required? My initial thought might be that `Hostname` and `Path` (soon to be `Address`) are the only two that might be considered required, but I'm not sure of a scenario where all fields wouldn't be set and couldn't be validated.

Reply:
That's a great question that I'm also uncertain what the answer should be 😛 

I think what you have initially would be good (require `Hostname` and `Address`) and we can always expand upon it later!

I too often find myself going through the mental gymnastics of trying to determine what should and shouldn't be considered a required field. The way I try to think about the required fields in a struct is more of a functionality perspective and what is actually necessary to perform actions within Vela.

I know that's a bit vague so to be more helpful, let's apply context from this specific situation to better illustrate 😃 

From a functionality perspective, I think to myself:

> What fields in the database table will actually be used in the code?

To answer, I imagine the `Address` is absolutely necessary because that's the value we'll use to send API requests from the server to the worker 👍 

Since the `Address` itself will also contain the `Hostname`, I think it's reasonable to also say that would be required.

As we start to evaluate the other fields, I'm unsure we can apply the same "required" label to it but that doesn't mean those fields aren't important or have a purpose.

For example, would `LastCheckedIn` or `Active` be things that are actually used in the code?

The answer is probably not although they are still useful and serve a purpose, especially from an admin perspective 👍 

And to your point, in a perfect world, **all fields** should be set and could be validated, but for now, I'd recommend only validating what is absolutely required 😄

Reply:
thanks for the explanation! I could see `Active` being used when the server goes to connect to a worker. It might check to see that the worker is marked as active before attempting to connect and waiting for a timeout, but other than that I don't see those being used currently.

I'll add the Address and Hostname fields for now. Like you mentioned we can always add more so I'll get  the function in place to make that easier.

Reply:
That's a fair point 👍 

We do use the `Active` field in other parts of the codebase to make sure it is `true` before proceeding with more actions.

But sounds like you got it figured out and know where to go from here! 🎉 

If you have more questions let me know otherwise I'll be on the lookout for more changes 👀